src/Camunda8/Services/OAuthToken/AccessToken.cs

```diff
@@ -0,0 +1,31 @@
+
+using Newtonsoft.Json.Linq;
+
+
+public class AccessToken(string Token, long DueDate)
+{
+
+    public string Token { get; } = Token;
+    public long DueDate { get; } = DueDate;    
+    public override string ToString()
+    {
+        return $"{nameof(Token)}: {Token}, {nameof(DueDate)}: {DueDate}";
+    }
+
+    public static AccessToken FromJson(string result)
+    {
+        var jsonResult = JObject.Parse(result);
+        var accessToken = (string)jsonResult["access_token"]!;
+
+        var expiresInMilliSeconds = (long)jsonResult["expires_in"]! * 1_000L;
+        var dueDate = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() + expiresInMilliSeconds;
```

Comment:
Do you think it will be easier to work with this in the future as a `long`, or as a `DateTime`? I am guessing you're hanging on to this for the purposes of caching/auto-requesting a new token. I wonder if you'll be writing long-to-DateTime conversion logic in those places, in which case it might make sense to capture it here as a DateTime, to avoid sprinkling the conversion logic in multiple places.

Reply:
This code I lifted verbatim from Zell's C# Zeebe package. I added the `isExpired()` method to encapsulate the implementation details, so it doesn't leak into the rest of the lib.